sonarqube/iap_proxy.py

```diff
@@ -16,18 +18,23 @@
 
 globalLogBeginner.beginLoggingTo([textFileLogObserver(stderr)])
 
-# Authenticate using service principal w/ key.
-def get_iap_access_token(aad_tenant, client_id, client_secret):
-    
-    authority_host_uri = 'https://login.microsoftonline.com'
-    authority_uri = authority_host_uri + '/' + aad_tenant
-    resource_uri = 'https://management.core.windows.net/'
-
-    context = adal.AuthenticationContext(authority_uri, api_version=None)
-    mgmt_token = context.acquire_token_with_client_credentials(resource_uri, client_id, client_secret)
-    accessToken = mgmt_token.get("accessToken")
-
-    return accessToken
+def get_oidc_token(request, client_id, service_account):
+    sa_info = json.loads(service_account)
+    credentials = IDTokenCredentials.from_service_account_info(
+        sa_info, target_audience=client_id
+    )
+    credentials.refresh(request)
+    return credentials.token
+
+def exchange_google_id_token_for_gcip_id_token(api_key, google_open_id_connect_token):
+  SIGN_IN_WITH_IDP_API = 'https://identitytoolkit.googleapis.com/v1/accounts:signInWithIdp'    
+  url = SIGN_IN_WITH_IDP_API + '?key=' + api_key
+  data={'requestUri': 'http://localhost',
```

Comment:
Why is the requestUri localhost?

Reply:
According to the docs, this is the URI which the IDP redirects back to.